compiler/rustc_target/src/spec/abi.rs

```diff
@@ -300,21 +296,21 @@ impl Abi {
             RustCold => 34,
         };
         debug_assert!(
-            AbiDatas
+            AbiData
                 .iter()
                 .enumerate()
-                .find(|(_, AbiData { abi, .. })| *abi == self)
+                .find(|(_, AbiDatum { abi, .. })| *abi == self)
                 .map(|(index, _)| index)
                 .expect("abi variant has associated data")
                 == i,
-            "Abi index did not match `AbiDatas` ordering"
+            "Abi index did not match `AbiData` ordering"
         );
         i
     }
 
     #[inline]
-    pub fn data(self) -> &'static AbiData {
-        &AbiDatas[self.index()]
+    pub fn data(self) -> &'static AbiDatum {
```

Comment:
The scope of this PR is way too wide. Renaming structs in the compiler shouldn't be mixed with fixing comments.

Reply:
I'll split it into separate PRs for compiler/library/rustdoc as @Nilstrieb mentioned